e2e_testing/torchscript/histogram_binning_calibration.py

```diff
@@ -0,0 +1,103 @@
+        self._iteration = 0
+
+    @export
+    @annotate_args([
+        None,
+        ([5000], torch.int32, True),
+        ([5001], torch.int32, True),
+        ([5000], torch.float32, True),
+    ])
+    def forward(self, segment_value, segment_lengths, logit):
+        origin_prediction = torch.sigmoid(
+            logit + torch.log(self.positive_weight))
+        dense_segment_value = torch.zeros(logit.numel(), dtype=torch.int32)
+        validoffsets = torch.gt(
+            segment_lengths[1:self._num_bins+1], segment_lengths[0:self._num_bins])
+        gathered_segment_values = (
+            segment_value[segment_lengths[0:self._num_bins].long()]+1).int()
+        dense_segment_value = torch.where(
+            validoffsets, gathered_segment_values, dense_segment_value)
+        zeros = torch.empty_like(
+            dense_segment_value, dtype=torch.int32).fill_(0)  # ConstantFill
+        isnotvalid = torch.gt(dense_segment_value, self._num_segments)  # GT
+        dense_segment_value = torch.where(
+            isnotvalid, zeros, dense_segment_value)  # Conditional
+        bin_ids_data = torch.ceil(origin_prediction/self.step)-1  # Bucketize
+        bin_ids_data = bin_ids_data.long()  # Cast
+        curr_segment_value = dense_segment_value * self._num_bins  # Mul
+        bin_ids_data2 = bin_ids_data  # Bucketize
+        bin_ids_data = bin_ids_data + curr_segment_value  # Add
+        curr_segment_value = self._bin_num_positives[bin_ids_data]  # Gather
+        curr_bin_num_examples = self._bin_num_examples[bin_ids_data]  # Gather
+        curr_segment_value = curr_segment_value / curr_bin_num_examples  # Div
+        curr_segment_value = curr_segment_value.float()  # Cast
+        curr_segment_value = curr_segment_value * self.bin_ctr_weight_value + \
+            origin_prediction * self.oneminusbin_ctr_weight_value  # WeightedSum
+        isvalid = torch.gt(curr_bin_num_examples,
+                           self.bin_ctr_in_use_after)  # GT
+        calibrated_prediction_data = torch.where(
+            isvalid, curr_segment_value, origin_prediction.float())  # Conditional
+        return calibrated_prediction_data, bin_ids_data
+
+
+@register_test_case(module_factory=lambda: HistogramBinningCalibrationByFeature())
+def HBC_staticsize(module, tu: TestUtils):
+    num_bins: int = 5000
+    num_segments: int = 42
+    num_logits = 5000
+    input_data_cpu = torch.rand(num_logits, dtype=torch.float)
+    segment_lengths: Tensor = torch.randint(
+        0, 2, (num_logits,), dtype=torch.int)
+    segment_offsets: Tensor = torch.cumsum(segment_lengths, 0)
+    segment_offsets: Tensor = torch.cat(
+        (torch.tensor([0]), segment_offsets), 0)
+    num_values: int = int(torch.sum(segment_lengths).item())
+    segment_values: Tensor = torch.randint(
+        0,
+        num_segments,
+        (num_values,),
+    )
+    segment_values = torch.cat(
+        (segment_values, torch.zeros(5000-segment_values.numel())), 0)

```

Comment:
nit: is this 5000 related to `num_bins` or `num_logits`? If so, it would be nice to use the variable